constraints/bsg_clk_gen/bsg_clk_gen.constraints.tcl

```diff
@@ -0,0 +1,53 @@
+puts "Info: Start script [info script]\n"
+
+############################################
+#
+# bsg_clk_gen timing assertions
+#
+proc bsg_clk_gen_clock_create { osc_path clk_name clk_gen_period_int clk_gen_period_ext osc_uncertainty ds_uncertainty clk_uncertainty} {
+  # very little is actually timed with this domain; just the receive
+  # side of the bsg_tag_client and the downsampler.
+  #
+  # Although the fastest target period of the oscillator itself is below
+  # this, we don't want this support logic to not be able to keep up
+  # in the event that oscillator runs faster than the tools say
+  #
+
+  # this is for the output of the downsampler, goes to the clock selection mux
+  set clk_gen_period_ds [expr $clk_gen_period_int * 2.0]
+
+  # this is for the output of the oscillator, which goes to the downsampler
+  create_clock -period $clk_gen_period_int -name ${clk_name}_osc [get_pins -leaf -of_objects [get_nets ${osc_path}osc_clk_out] -filter "pin_direction==out"]
+  set_clock_uncertainty $osc_uncertainty [get_clocks ${clk_name}_osc]
+
+  # it would be nice to remove the following version detection scripts if they are deprecated
+  # we always use version 2 of bsg_clk_gen in recent tapeouts
+  echo "Detecting Version 1/2 of bsg_clk_gen"
+  set buf_btc_o_search [sizeof_collection [get_pins ${osc_path}clk_gen_osc_inst/fdt/buf_btc_o]]
+  echo $buf_btc_o_search
+  # for version 1 bsg_clk_gen
+  if { $buf_btc_o_search } {
```

Comment:
is there a reason why we need two different versions of clk gen in the same module? why not just create a separate module?